### ISSUE TITLE
Support querying journal for field value ranges

### ISSUE THREAD

**Issue by lucaswerkmeister** [systemd/systemd#4091]
### Submission type
- [ ] Bug report
- [x] Request for enhancement (RFE)

I think it would be very useful if the journal and `journalctl` allowed you to query not only for specific `FIELD=VALUE` matches, but also for comparisons of the form `FIELD<=VALUE`, `FIELD>=VALUE`, and `FIELD=VALUE1..VALUE2`. Currently, something like this is only supported for the priority with `--priority FROM..TO`, and for the date with `--since` and `--until`, but this could also be useful for many application-defined fields.

This is probably most useful for numeric fields, but it’s probably easiest to only support a single sort specification, and in this case, lexical sort is probably the most generally useful option. Application developers should then make sure to pad their numeric values to constant length so that lexical sort is equivalent to numeric sort for those fields.

This may be impossible to support efficiently depending on how the journal indexes fields. I assume that it sorts them into a tree and then queries with a binary search; if hashing is used instead, feel free to close this issue.

**Comment by poettering** [comment]
Yeah, we currently use hashing, not a tree data structure. This can't be implemented effectively with the current journal format.

The priority match only works because the range for it is so small, hence we just do multiple lookups for all vlaues in the range.
